### PR TITLE
Search Settings #148

### LINKED ISSUE #148: Search Settings
Lets add two more Search settings in the Settings screen.

Following the implementation from Cliqz 2.0, we want to add language chooser and "Block explicit Content" settings.

IMG_4562

To get/set the language use `Search.getBackendCountries` and `Search.setBackendCountry`. The API work with language code which we have to translate to country names. The `getBackendCountries` can be updated to get the translated country name, but we can use our own translations instead.

To control explicit results we have to add two more static methods to `Search`: `getAdultFilter` and `setAdultFilter` they will return/accept String representing two states `liberal` for off and `conservative` for on. By default we block adult results.

(note that we need browser-core update to get `getAdultFilter` to work - @chrmod will take care of it)

### PR DESCRIPTION
**Pull request by shahenAnt**
## Ticket
fix #148 

## Description
Added search result for country settings.
Added Block explicit content settings.

## Checklist:

- [ ] My squashed commit messages follows the [7 golden rules](https://chris.beams.io/posts/git-commit/)
- [ ] My code follows the code style of this project.
- [ ] I updated or created necessary unit tests
- [ ] I updated all necessary documentation

### CODE REVIEW

File: UserAgent/Search.swift [hunk 1]
```diff
@@ -7,68 +7,105 @@
...
-                callback(Self.defaultConfig)
+                DispatchQueue.main.async {
+                    callback(Self.defaultConfig)
+                }
                 return
             }
 
             guard let backends = result as? [String: [String: Any]] else {
-                callback(Self.defaultConfig)
+                DispatchQueue.main.async {
```
Comment (chrmod) [inline UserAgent/Search.swift]: Would be better for the callback provider to deal the the queue.

Comment (shahenAnt) [reply UserAgent/Search.swift]: Done

File: Translations/de.lproj/Localizable.strings [hunk 2]
```diff
@@ -838,6 +838,15 @@
 /* The placeholder for URL Field when saving a custom search engine */
 "Settings.AddCustomEngine.URLPlaceholder" = "URL (Anfrage durch %s ersetzen)";
 
+/* The button text in Search Settings that opens the list of supported search languages. */
+"Settings.SearchResultForLanguage" = "";
```
Comment (pavel-cliqz) [inline Translations/de.lproj/Localizable.strings]: @when we should add the localizations. @chrmod  maybe we ask someone now?

Comment (shahenAnt) [reply Translations/de.lproj/Localizable.strings]: Done

**Pull request merged**